consensus/src/consensus/remote_event_dispatcher.rs

```diff
@@ -0,0 +1,329 @@
+use beserial::{Deserialize, Serialize};
+use futures::stream::BoxStream;
+use futures::StreamExt;
+use nimiq_hash::{Blake2bHash, Hash};
+use nimiq_network_interface::network::{NetworkEvent, SubscribeEvents, Topic};
+use parking_lot::RwLock;
+use std::collections::{HashMap, HashSet};
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use nimiq_blockchain::Blockchain;
+use nimiq_blockchain_interface::AbstractBlockchain;
+use nimiq_blockchain_interface::BlockchainEvent;
+use nimiq_keys::Address;
+use nimiq_network_interface::request::Handle;
+use nimiq_network_interface::{network::Network, request::request_handler};
+
+use crate::messages::RequestSubscribeToAddress;
+use crate::messages::{AddressSubscriptionOperation, ResponseSubscribeToAddress};
+use crate::SubscribeToAdressesError::*;
+
+//The max number of peers that can be subscribed.
+pub const MAX_SUBSCRIBED_PEERS: usize = 5;
+//The max number of addresses that can be subscribed, per peer
+pub const MAX_SUBSCRIBED_PEERS_ADDRESSES: usize = 10;
+
+/// Different kind of events that could generate notifications
+#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
+#[repr(u8)]
+pub enum NotificationEvent {
+    BlockchainExtend,
+}
+
+/// Interesting Addresses Notifications:
+/// A collection of transaction receipts that might be interesting for some peer
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct AddressNotification {
+    /// The Event that generated this notification
+    pub event: NotificationEvent,
+    /// Tuples of `(transaction_hash, block_number)`
+    #[beserial(len_type(u16, limit = 128))]
+    pub receipts: Vec<(Blake2bHash, u32)>,
+}
+
+/// Topic used to notify peers about transaction adddresses they are subscribed to
+/// The final notification is sent over a subtopic derived from this one, which is specific to each peer
+#[derive(Clone, Debug, Default)]
+pub struct AddressSubscriptionTopic;
+
+impl Topic for AddressSubscriptionTopic {
+    type Item = AddressNotification;
+
+    const BUFFER_SIZE: usize = 1024;
+    const NAME: &'static str = "transactions";
```

Comment:
Maybe rename it to avoid possible conflicts with the regular `transaction` topic